### PR TITLE
Add implementation for calculating Earned PSP

### PR DESCRIPTION
**Pull request by shresthagrawal**

### CODE REVIEW

**Review by nduchak — APPROVED** [review]

**Review by mwamedacen — APPROVED** [review]

File: src/lib/pool-info.ts [hunk 1]
```diff
@@ -358,6 +377,85 @@ export class PoolInfo {
...
+    const multiCallData = pools.map(p => {
+      return {
+        target: p.address,
+        callData: this.erc20Interface.encodeFunctionData('balanceOf', [
+          userAddress,
+        ]),
+      };
+    });
+
+    const rawResult = await this.multicallContract.functions.aggregate(
```
Comment (mwamedacen) [inline src/lib/pool-info.ts:423]: we can memoize this too, on memory for now

**Review by nduchak — APPROVED** [review]
LGTM

File: src/lib/pool-info.ts (hunk 1 above)
Comment (shresthagrawal) [reply src/lib/pool-info.ts:423]: An attacker can simply call with random address which will bombard the memory

Comment (mwamedacen) [reply src/lib/pool-info.ts:423]: Very fair point

**Pull request merged**